### ISSUE TITLE
[QUESTION] Registered CommandHandler can not be entry_point for ConversationHandler.

### ISSUE THREAD

**Issue by manuelfedele** [python-telegram-bot/python-telegram-bot#1942]
I know that registering a CommandHandler to dispatcher AND to ConversationHandler entry_points is stupid, but this took me some time to understand the issue. I'll explain why.

#### The "Problem".
```python
def start(update, context):
    # do something
    return 1

def one(update, context):
    # do something else


conv_handler = ConversationHandler(
    entry_points = [ComandHandler('start', start)]
    states = { 1: CallbackQueryHandler(one) }
    fallbacks = [ComandHandler('start', start)]
)

updater = Updater(<token>)
updater.dispatcher.add_handler(ComandHandler('start', start))
updater.dispatcher.add_handler(conv_handler)
updater.start_polling()
```

This won't work. ConversationHandler will always get `state` None, and `one` will never be called.

#### Describe the solution you'd like
Even if I understand that the code described above makes not much sense, while refactoring a very large bot  this can happen, and is very difficult do debug, because:
1) This is undocumented.
2) The logger, even in debug mode, does not provide much more information except `"selecting conversation <id> with state None"`

#### Describe alternatives you've considered
1) Point this in documentation.
2) Raise an exception if someone tries to register the same handler to pure dispatcher handlers **and** to a ConversationHandler's entry_points

**Comment by Bibo-Joshi** [comment]
Citing from the documentation of [`Dispatcher.add_handler`](https://python-telegram-bot.readthedocs.io/en/stable/telegram.ext.dispatcher.html#telegram.ext.Dispatcher.add_handler):

> TL;DR: Order and priority counts. 0 or 1 handlers per group will be used. End handling of update with `telegram.ext.DispatcherHandlerStop`.

Personally, I think it's pretty clear from that, that a conversation can only start, if it's possible for an update to reach an entry point handler. While it may be a bit more complicated to overlook your handlers for a large bot, the logic behind that is very simple: If an update is handled in a group, no other handlers in that group will be called.

**TL;DR:** I don't see, that anything would need improvement. Or am I misunderstanding you?

**Comment by manuelfedele** [comment]
I admit to have little knowledge of this library (just used on 2 projects) but in my tests, changing order, like this (refer to code above):

```python
updater.dispatcher.add_handler(conv_handler)
updater.dispatcher.add_handler(ComandHandler('start', start))
```
produces the same result. `conv_handler` still has lower priority and its entry_point is never reached.

So what exactly does this mean?

> TL;DR: Order and priority counts. 0 or 1 handlers per group will be used. End handling of update with telegram.ext.DispatcherHandlerStop.

**Comment by Bibo-Joshi** [comment]
>  `conv_handler` still has lower priority and its entry_point is never reached.

Then you are doing something wrong. Minimal working example:
 ```python
#!/usr/bin/env python
# -*- coding: utf-8 -*-

from telegram.ext import Updater, CommandHandler, ConversationHandler

def start(update, context):
    update.message.reply_text('This is the CommandHandler listening to /start.')


def start_conversation(update, context):
    update.message.reply_text('This is the conversations entry point listening to /start.')
    return 1


def state(update, context):
    update.message.reply_text('This is the conversations only state listening to /cmd.')
    return ConversationHandler.END


def main():
    updater = Updater("703777048:AAH4G0IMpY1vLh_S2UbbZgBqyIT26_lXNYk", use_context=True)

    dp = updater.dispatcher

    dp.add_handler(ConversationHandler(
        entry_points=[CommandHandler('start', start_conversation)],
        states={1: [CommandHandler('cmd', state)]},
        fallbacks=[],
    ))
    dp.add_handler(CommandHandler('start', start))

    updater.start_polling()
    updater.idle()


if __name__ == '__main__':
    main()

```

Sending `/start` starts the conversation as expected and `start()` is never called. If you cange

```python
 dp.add_handler(CommandHandler('start', start))
```
to
```python
 dp.add_handler(CommandHandler('start', start), 1)
```
the `CommandHandler` is in a different group than the `ConversationHandler` and will indeed be called, as expected.
